lib/quantization/src/encoded_vectors_binary.rs

```diff
@@ -212,17 +314,79 @@ impl<TBitsStoreType: BitsStoreType, TStorage: EncodedStorage>
                 encoded_vector[i / bits_count] |= one << (i % bits_count);
             }
         }
+    }
 
-        EncodedBinVector { encoded_vector }
+    fn encode_two_bits_vector(
+        vector: &[f32],
+        encoded_vector: &mut [TBitsStoreType],
+        standard_deviations: &[f32],
+        means: &[f32],
+    ) {
+        let bits_count = u8::BITS as usize * std::mem::size_of::<TBitsStoreType>();
+        let one = TBitsStoreType::one();
+        for (i, &v) in vector.iter().enumerate() {
+            let mean = means[i];
+            let sd = standard_deviations[i];
+            let ranges = 3;
+            let v_z = (v - mean) / sd;
+            let index = (v_z + 2.0) / (4.0 / ranges as f32);
```

Comment:
this part requires a comment. Where 2.0 and 4.0 is coming from?

Reply:
I did a refactor, see `fn encode_two_bits_value(value: f32, mean: f32, sd: f32)`. It has comments for each step and explanes all magic numbers